src/helper/bit-tools/oval-tool.js

```diff
@@ -57,17 +57,49 @@ class OvalTool extends paper.Tool {
      */
     onSelectionChanged (selectedItems) {
         this.boundingBoxTool.onSelectionChanged(selectedItems);
-        if ((!this.oval || !this.oval.parent) &&
+        if ((!this.oval || !this.oval.isInserted()) &&
                 selectedItems && selectedItems.length === 1 && selectedItems[0].shape === 'ellipse') {
             // Infer that an undo occurred and get back the active oval
             this.oval = selectedItems[0];
-        } else if (this.oval && this.oval.parent && !this.oval.selected) {
```

Comment:
What is the change to the if logic involving `isInserted` do?

Reply:
I'm not sure how, but I hit an issue where the oval had a parent, but it was a group that had no parent, so it wasn't attached to the view. I discovered isInserted in the API and it's exactly what I wanted, that the item is in the view. I might go through later and change more occurrences of parent !== null to isInserted.

Reply:
ah cool